### PR TITLE
Gpl 263a pre extension time

### PR DESCRIPTION
**Pull request by KatyTaylor**
Still waiting to:
- hear back from Craig RE a couple of questions _- done_
- test this manually with traction ui branch once ready  _- done_

### CODE REVIEW

File: spec/requests/v1/pacbio/wells_spec.rb [hunk 1]
```diff
@@ -127,7 +129,42 @@
...
+                      }
+                    }
+                  }
+                ],
+              }
+            }
+          }.to_json
+        end
 
+        it 'creates a well with pre-extension time set' do
```
Comment (harrietc52) [inline spec/requests/v1/pacbio/wells_spec.rb:163]: Could be worth testing with more than one well in the `wells` body, to make sure different wells with different CCS is inserted correctly

File: app/models/concerns/sample_sheet.rb [hunk 2]
```diff
@@ -40,4 +40,10 @@ def same_barcodes_on_both_ends_of_sequence
     # Always true at the time of writing
     true
   end
+
+  def automation_parameters
+    return unless pre_extension_time
+
+    "ExtensionTime=double:#{pre_extension_time}|ExtendFirst=boolean:True"
```
[the remark this answers is not in the document]
Comment (KatyTaylor) [reply app/models/concerns/sample_sheet.rb:47]: Can't find anything helpful in the user guide. I'll mention it to Craig when I email about UAT.

File: spec/models/pacbio/well_spec.rb [hunk 3]
```diff
@@ -92,6 +91,17 @@
     end
   end
 
+  context 'pre-extension time' do
+    it 'is not required' do
+      expect(build(:pacbio_well).pre_extension_time).to be_nil
```
[the remark this answers is not in the document]
Comment (KatyTaylor) [reply spec/models/pacbio/well_spec.rb:96]: Hmm I actually added it before I added the validation line, but that's a good idea, will add a `create`. Thanks

File: spec/requests/v1/pacbio/wells_spec.rb (hunk 1 above)
Comment (KatyTaylor) [reply spec/requests/v1/pacbio/wells_spec.rb:163]: Sure can do. What do you mean `with different CCS` - do you mean with different sequencing mode, or different pre-extension time? Thanks

**Pull request merged**